lib/spack/spack/relocate.py

```diff
@@ -415,6 +426,17 @@ def make_binary_placeholder(cur_path_names, allow_root):
         tty.die("Placeholder not implemented for %s" % platform.system())
 
 
+def relocate_link(path_names, old_dir, new_dir):
+    """
+    Replace old path with new path in link sources
+    """
+    for path_name in path_names:
+        old_src = os.path.realpath(path_name)
+        new_src = old_src.replace(old_dir, new_dir, 1)
```

Comment:
If `old_dir` here is the old Spack prefix, I'll note that `binary_distribution.build_tarball` creates a tarball in a temp directory; it uses `llnl.util.filesystem.install_tree` which relocates absolute symlinks, so at that point, all the absolute symlinks are defined WRT the temp directory as the root.

I bring this up because I think the additional tests are insufficient. Is it possible to create a test that invokes `build_tarball`?